sway/tree/view.c

```diff
@@ -463,9 +464,23 @@ void view_map(struct sway_view *view, struct wlr_surface *wlr_surface) {
 		return;
 	}
 
+	pid_t pid;
+	if (view->type == SWAY_VIEW_XWAYLAND) {
+		struct wlr_xwayland_surface *surf =
+			wlr_xwayland_surface_from_wlr_surface(wlr_surface);
+		pid = surf->pid;
+	} else {
+		struct wl_client *client =
+			wl_resource_get_client(wlr_surface->resource);
+		uid_t uid;
+		gid_t gid;
+		wl_client_get_credentials(client, &pid, &uid, &gid);
```

Comment:
You can use NULL for uid and gid if you're not interested in their values:
https://github.com/wayland-project/wayland/blob/0e6ac72288f884ba0321c285a65578492c706534/src/wayland-server.c#L563-L564